### PR TITLE
DCT deprecation note

### PR DESCRIPTION
**Pull request by amyblais**
https://hub.mattermost.com/private-core/pl/mgwwcy4yi3fobrz7bfxdztg6ua

Forum post to be drafted as well.

### CODE REVIEW

File: source/about/deprecated-features.rst [hunk 1]
```diff
@@ -11,6 +11,11 @@ Mattermost Server v10.3.0
 
 - As part of our continuous efforts to enhance your user experience, we will be phasing out the Classic Mobile App. Please download the new v2 Mobile App from the `Apple App Store <https://apps.apple.com/us/app/mattermost/id1257222717>`_ or `Google Play Store <https://play.google.com/store/apps/details?id=com.mattermost.rn>`_ to continue enjoying our services without interruption. See more details in `this forum post <https://forum.mattermost.com/t/classic-mobile-app-deprecation/18703>`_.
 
+Mattermost Server v10.2.0
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Docker Content Trust (DCT) for signing Docker image artifacts will be replaced by Sigstore CoSign in our upcoming release, v10.2 (November, 2024). Please see details `here <https://docs.sigstore.dev/system_config/installation/>`_ on installing CoSign.
```
Comment (saturninoabril) [inline source/about/deprecated-features.rst:17]: I believe the name they're using is `Cosign` and not `CoSign`.

```suggestion
- Docker Content Trust (DCT) for signing Docker image artifacts will be replaced by Sigstore Cosign in our upcoming release, v10.2 (November, 2024). Please see details `here <https://docs.sigstore.dev/system_config/installation/>`_ on installing Cosign.
```

File: source/upgrade/important-upgrade-notes.rst [hunk 2]
```diff
@@ -13,6 +13,7 @@ Important Upgrade Notes
    - Mattermost plugins built with Go versions 1.22.0 and 1.22.1 do not work. Plugin developers should use go 1.22.2 or newer instead.
    - Keybase has stopped serving our Ubuntu repository signing key. If you were using it, update your installation scripts to retrieve the key as mentioned in our docs: https://docs.mattermost.com/install/install-ubuntu.html.
    - Mattermost v10.0 is planned for September, 2024. See a list of planned breaking changes for this release `in the changelog <https://docs.mattermost.com/about/mattermost-v10-changelog.html>`__.
+   - Docker Content Trust (DCT) for signing Docker image artifacts will be replaced by Sigstore CoSign in our upcoming release, v10.2 (November, 2024). Please see details `here <https://docs.sigstore.dev/system_config/installation/>`_ on installing CoSign.
```
Comment (saturninoabril) [inline source/upgrade/important-upgrade-notes.rst:16]:
```suggestion
   - Docker Content Trust (DCT) for signing Docker image artifacts will be replaced by Sigstore Cosign in our upcoming release, v10.2 (November, 2024). Please see details `here <https://docs.sigstore.dev/system_config/installation/>`_ on installing Cosign.
```

**Review by saturninoabril — APPROVED** [review]
Thanks @amyblais! Left comment on name change.

**Comment by phoinixgrr** [comment]
I would remove the COSIGN installation instructions. This should be a change log. 

We should better link to an internal documentation page, showing the migration steps to use Cosign  (installation, sign check)

**Comment by phoinixgrr** [comment]
Providing some more context here, 

Additionally,  we already have everything signed with DCT and COSIGN.  We will discontinue signing with DCT as of a specific date/release. Therefore, if you(client) rely on artifact verification using DCT, now is the time to transition to COSIGN.

**Review by cwarnermm — APPROVED** [review]

**Review by phoinixgrr — APPROVED** [review]

**Comment by amyblais** [comment]
> We should better link to an internal documentation page, showing the migration steps to use Cosign (installation, sign check) , that clearly states, that if you utilize DCT for container image verification, then and only then you are affected.

@phoinixgrr I'm thinking of linking to the forum post once it's published. Would this be ok, or did you have something else in mind for the "internal documentation page"?

**Comment by phoinixgrr** [comment]
> > We should better link to an internal documentation page, showing the migration steps to use Cosign (installation, sign check) , that clearly states, that if you utilize DCT for container image verification, then and only then you are affected.
> 
> @phoinixgrr I'm thinking of linking to the forum post once it's published. Would this be ok, or did you have something else in mind for the "internal documentation page"?

Works for me

File: source/about/deprecated-features.rst [hunk 3]
```diff
@@ -11,6 +11,11 @@ Mattermost Server v10.3.0
 
 - As part of our continuous efforts to enhance your user experience, we will be phasing out the Classic Mobile App. Please download the new v2 Mobile App from the `Apple App Store <https://apps.apple.com/us/app/mattermost/id1257222717>`_ or `Google Play Store <https://play.google.com/store/apps/details?id=com.mattermost.rn>`_ to continue enjoying our services without interruption. See more details in `this forum post <https://forum.mattermost.com/t/classic-mobile-app-deprecation/18703>`_.
 
+Mattermost Server v10.2.0
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Docker Content Trust (DCT) for signing Docker image artifacts will be replaced by Sigstore Cosign in our upcoming release, v10.2 (November, 2024). If you rely on artifact verification using DCT, please transition to using Cosign.
```
Comment (nab-77) [inline source/about/deprecated-features.rst:17]: @amyblais do we have a link to a resource (maybe from Docker themselves) on how to transition to Cosign?

@cwarnermm as always favour a note-box and link-out on this guidance.

**Review by nab-77 — CHANGES_REQUESTED** [review]
Question on external resources to assist in transition.

File: source/about/deprecated-features.rst (hunk 3 above)
Comment (amyblais) [reply source/about/deprecated-features.rst:17]: @nab-77 I will add a link to the forum post once published https://docs.google.com/document/d/1ro_skgRqtDVzul_ukmkgu23SgLtdBl4EQsPZnQIho5Y/edit. I'll add a link to the docs as well https://edu.chainguard.dev/open-source/sigstore/cosign/how-to-install-cosign/.

**Review by nab-77 — APPROVED** [review]
Superb stuff @amyblais

**Pull request merged**